pym/portage/emaint/modules/logs/logs.py

```diff
@@ -48,16 +47,15 @@ def clean(self, **kwargs):
 		"""
 		num_of_days = None
 		pretend = False
-		if kwargs:
-			# convuluted, I know, but portage.settings does not exist in
-			# kwargs.get() when called from _emerge.main.clean_logs()
-			settings = kwargs.get('settings', None)
-			if not settings:
-				settings = portage.settings
-			options = kwargs.get('options', None)
-			if options:
-				num_of_days = options.get('NUM', None)
-				pretend = options.get('pretend', False)
+
+		# convoluted, I know, but portage.settings does not exist in
+		# kwargs.get() when called from _emerge.main.clean_logs()
+		settings = kwargs.get('settings', portage.settings)
```

Comment:
This reference to `portage.settings` should raise AttributeError when called by emerge, since emerge calls `portage._disable_legacy_globals()`. This should suppress the AttributeError:
```
settings = kwargs.get('settings', getattr(portage, 'settings', None))
```